src/accountant.rs

```diff
@@ -131,23 +133,34 @@ impl Accountant {
         // Hold a write lock before the condition check, so that a debit can't occur
         // between checking the balance and the withdraw.
         let option = bals.get(&tr.from);
+
         if option.is_none() {
             return Err(AccountingError::AccountNotFound);
         }
-        let mut bal = option.unwrap().write().unwrap();
 
         if !self.reserve_signature_with_last_id(&tr.sig, &tr.data.last_id) {
             return Err(AccountingError::InvalidTransferSignature);
         }
 
-        if *bal < tr.data.tokens {
+        let bal = option.unwrap();
+        let current = bal.load(Ordering::Relaxed) as i64;
+
+        if current < tr.data.tokens {
             self.forget_signature_with_last_id(&tr.sig, &tr.data.last_id);
             return Err(AccountingError::InsufficientFunds);
         }
 
-        *bal -= tr.data.tokens;
+        let result = bal.compare_exchange(
+            current as isize,
+            (current - tr.data.tokens) as isize,
+            Ordering::Relaxed,
+            Ordering::Relaxed,
+        );
 
-        Ok(())
+        return match result {
+            Ok(_) => Ok(()),
```

Comment:
If you don't mind, please run Clippy before submitting PRs. It would have a suggestion to make this code more idiomatic.